src/bet.rs

```diff
@@ -344,5 +354,333 @@ pub(crate) async fn generate_toasts(
+        data: GameData,
+    }
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct GameData {
+        id: String,
+        away_pitcher: String,
+        home_pitcher: String,
+    }
+
+    #[derive(PartialEq, Debug)]
+    enum InningHalf {
+        Top,
+        Bottom,
+    }
+
+    impl std::ops::Not for InningHalf {
+        type Output = Self;
+
+        fn not(self) -> Self::Output {
+            match self {
+                InningHalf::Top => InningHalf::Bottom,
+                InningHalf::Bottom => InningHalf::Top,
+            }
+        }
+    }
+
+    let games: Vec<GameResult> = RequestBuilder::v1("games")
+        .after(after - Duration::hours(1)) // this is a hack to catch games going on right now. it's bad and needs a fix - allie
+        .before(before)
+        .pitcher(pitcher.to_string())
+        .finished(if only_finished_games {
+            Some(true)
+        } else {
+            None
+        })
+        .json(config)
+        .await?
+        .data;
+
+    Ok(stream! {
+        // yield all events for a game at once at once, so we can do per-game counts like shut outs
+        let mut events: Vec<MinimalFeedEvent> = Vec::new();
+
+        for game in games {
+            // todo: support event type 3, 'pitcher change'
+            let pitcher_team = if game.data.away_pitcher == pitcher { InningHalf::Top } else { InningHalf::Bottom };
+            // hack to get all events at once. impl paging?
+            let mut inning_half = InningHalf::Top;
+
+            for event in feed::all_events_for_game::<MinimalFeedEvent>(config, &game.data.id, before, after, "10000000").await? {
+                if event.etype == 2 {
+                    inning_half = !inning_half;
+                }
+
+                if inning_half == pitcher_team {
+                    events.push(event);
+                }
+            }
+
+            yield Ok(events.drain(..).collect());
+        }
     })
 }
+
+/// gets triggers for events that can be counted from feed events
+async fn event_based_snack_triggers(
+    config: &Config,
+    cookies: &CookieJar<'_>,
+    last_snack_payout: DateTime,
+    from: DateTime,
+    to: DateTime,
+    payout_info: &SnackPayoutInfo,
+) -> anyhow::Result<usize> {
+    feed::count_events(
+        config,
+        &payout_info.event_types,
+        from,
+        to,
+        payout_info.filters.iter().map(|(k, v)| {
+            (
+                (*k).to_string(),
+                v.replace(
+                    "::IDOL::",
+                    &cookies
+                        .load::<Idol>()
+                        .map(|v| v.to_string())
+                        .unwrap_or_default(),
+                )
+                .replace(
+                    "::FAV_TEAM::",
+                    &cookies
+                        .load::<FavoriteTeam>()
+                        .map(|v| v.to_string())
+                        .unwrap_or_default(),
+                ),
```

Comment:
Could the filter values be some kind of enum instead of doing string matching/replacement, or is there a use case you're preparing for down the road?

Reply:
oh, good call